File: Year-2019/Mar/2019-03-12-242/watercow/387.c
class Solution {
public:
	int firstUniqChar(string s) {
		vector<int> count(26, 0);

		for (auto i : s) ++count[i - 'a'];

		for (int i = 0; i < s.size(); i++) {
			if (count[s[i] - 'a'] == 1) {
				return i;
			}
		}

		return -1;
	}
};

File: Year-2019/Mar/2019-03-12-242/lb/387.java
class Solution {
    public int firstUniqChar(String s) {
        int[] si = new int[26];
        char[] sc = s.toCharArray();
        for (char c : sc){
            si[c-'a'] ++;
        }
        for (int i = 0;i<sc.length;i++){
            if (si[sc[i] - 'a'] == 1)
                return i;
        }
        return -1;
    }
}
